src/jdk.incubator.code/share/classes/jdk/incubator/code/TypeElement.java

```diff
@@ -86,6 +86,42 @@ static int dimensions(String identifier) {
 
         // Factories
 
+        public static ExternalizedTypeElement of(String s) {
+            return new ExternalizedTypeElement(s, List.of());
+        }
+
+        public static ExternalizedTypeElement of(String s,
+                                                 ExternalizedTypeElement a) {
+            return new ExternalizedTypeElement(s, List.of(a));
+        }
+
+        public static ExternalizedTypeElement of(String s,
+                                                 ExternalizedTypeElement a1, ExternalizedTypeElement a2) {
+            return new ExternalizedTypeElement(s, List.of(a1, a2));
+        }
+
+        public static ExternalizedTypeElement of(String s,
+                                                 ExternalizedTypeElement a1, ExternalizedTypeElement a2,
+                                                 ExternalizedTypeElement a3) {
+            return new ExternalizedTypeElement(s, List.of(a1, a2, a3));
+        }
+
+        public static ExternalizedTypeElement of(String s,
+                                                 ExternalizedTypeElement a1, ExternalizedTypeElement a2,
+                                                 ExternalizedTypeElement a3, ExternalizedTypeElement a4) {
+            return new ExternalizedTypeElement(s, List.of(a1, a2, a3, a4));
+        }
```

Comment:
It's possible we don't need these methods. They were included so that it was more efficient to encode ex type elem construction in the class file. Since instances used more than once are shared the small incremental cost of using the list accepting method is likely a good tradeoff.

(General idea - we should annotate methods that are referenced by the compiler and runtime and used for generating and producing code models, such that we know it will cause issues if we remove or refactor them.)